packages/web/components/trade-clipboard/index.tsx

```diff
@@ -0,0 +1,481 @@
+      .getBalanceFromCurrency(tradeTokenInConfig.sendCurrency);
+
+    const [isSettingOpen, setIsSettingOpen] = useBooleanWithWindowEvent(false);
+
+    const tokenSelectCurrencies = useMemo(() => {
+      return tradeTokenInConfig.sendableCurrencies.map((cur) => {
+        if ("originChainId" in cur) {
+          const originChainId = cur.originChainId;
+          if (originChainId && chainStore.hasChain(originChainId)) {
+            const chainInfo = chainStore.getChain(originChainId);
+            return {
+              ...cur,
+              meta: [chainInfo.chainName],
+            };
+          }
+          return cur;
+        }
+
+        return {
+          ...cur,
+          meta: [chainStore.osmosis.chainName],
+        };
+      });
+    }, [chainStore, tradeTokenInConfig.sendableCurrencies]);
+
+    const tokenInSelectCurrencies = useMemo(() => {
+      return tokenSelectCurrencies.filter(
+        (cur) =>
+          cur.coinMinimalDenom !==
+          tradeTokenInConfig.outCurrency.coinMinimalDenom
+      );
+    }, [
+      tokenSelectCurrencies,
+      tradeTokenInConfig.outCurrency.coinMinimalDenom,
+    ]);
+
+    const tokenOutSelectCurrencies = useMemo(() => {
+      return tokenSelectCurrencies.filter(
+        (cur) =>
+          cur.coinMinimalDenom !==
+          tradeTokenInConfig.sendCurrency.coinMinimalDenom
+      );
+    }, [
+      tokenSelectCurrencies,
+      tradeTokenInConfig.sendCurrency.coinMinimalDenom,
+    ]);
+
+    const manualSlippageInputRef = useRef<HTMLInputElement | null>(null);
+
+    useEffect(() => {
+      if (isSettingOpen && slippageConfig.isManualSlippage) {
+        // Whenever the setting opened, give a focus to the input if the manual slippage setting mode is on.
+        manualSlippageInputRef.current?.focus();
+      }
+      // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [isSettingOpen]);
+
+    return (
+      <div
+        className={classNames(
+          "relative rounded-2xl bg-card border-2 border-cardInner p-2.5",
+          containerClassName
+        )}
+        style={containerStyle}
+        ref={forwardedRef}
+      >
+        <div className="rounded-xl bg-cardInner px-5 pt-5 pb-8">
+          <div className="absolute -top-2 inset-x-1/2 -translate-x-1/2 w-[10rem] h-[3.75rem] z-10 bg-gradients-clip rounded-md">
+            <div className="absolute bottom-0 rounded-b-md w-full h-5 bg-gradients-clipInner" />
+            <div className="absolute inset-x-1/2 -translate-x-1/2 bottom-2 w-12 h-[1.875rem] bg-[rgba(91,83,147,0.12)] rounded-md shadow-[rgba(0,0,0,0.25)_1px_1px_1px_inset]" />
+          </div>
+
+          <div className="relative flex justify-end w-full h-11 mb-[1.125rem]">
+            <button
+              className="relative"
+              onClick={(e) => {
+                e.stopPropagation();
+                setIsSettingOpen(!isSettingOpen);
+              }}
+            >
+              <img
```

Comment:
Should we use `next/image` here?

Reply:
I'll take a look to see if I could possibly change this.